File: TripKitAndroid/src/main/java/com/skedgo/geocoding/GCSkedgoResult.java
package com.skedgo.geocoding;


import com.skedgo.geocoding.agregator.GCSkedGoResultInterface;

import org.jetbrains.annotations.NotNull;

import java.util.List;

import androidx.annotation.Nullable;

public class GCSkedgoResult extends GCResult implements GCSkedGoResultInterface {

    @NotNull
//  class json field from skedgo's json
    private String resultClass;
    //  popularity json field from skedgo's json
    private int popularity;

    @Nullable
    private List<String> modeIdentifiers;

    public GCSkedgoResult(
        String name,
        double lat,
        double lng,
        @NotNull String resultClass,
        Integer popularity,
        @Nullable List<String> modeIdentifiers
    ) {
        super(name, lat, lng);
        this.popularity = popularity;
        this.resultClass = resultClass;
        this.modeIdentifiers = modeIdentifiers;
    }

    @NotNull
    @Override
    public String getResultClass() {
        return resultClass;
    }

    public void setResultClass(@NotNull String resultClass) {
        this.resultClass = resultClass;
    }

    @Override
    public int getPopularity() {
        return popularity;
    }

    public void setPopularity(int popularity) {
        this.popularity = popularity;
    }

    public boolean isStopLocation() {
        return this.resultClass.equalsIgnoreCase("StopLocation");
    }

    @Nullable
    @Override
    public List<String> getModeIdentifiers() {
        return modeIdentifiers;
    }
}

File: TripKitAndroid/src/main/java/com/skedgo/geocoding/GCSkedgoResult.kt
package com.skedgo.geocoding

import com.skedgo.geocoding.agregator.GCSkedGoResultInterface


class GCSkedgoResult(
    name: String?,
    lat: Double,
    lng: Double,
    //  class json field from skedgo's json
    override var resultClass: String,
    //  popularity json field from skedgo's json
    override var popularity: Int,
    override val modeIdentifiers: List<String>?
) : GCResult(name!!, lat, lng), GCSkedGoResultInterface {
    val isStopLocation: Boolean
        get() = resultClass.equals("StopLocation", ignoreCase = true)
}
